### PR TITLE
test: add TAS simple level and constraint tests

### PR DESCRIPTION
**Pull request by Ronkahn21**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR adds 5 test scenarios for simple topology constraint patterns in Topology Aware Scheduling (TAS).

**Tests Added:**
- **SL1_PCSOnlyConstraint**: Tests PCS-level constraint inherited by child PCSG and PCLQ resources
- **SL2_PCSGOnlyConstraint**: Tests constraint at PCSG level only (no parent PCS constraint)
- **SL3_NoTopologyConstraint**: Baseline test with no topology constraints applied
- **PC1_HostLevelConstraint**: Tests strictest host-level constraint (all pods on same host)
- **ZL1_ZoneLevelConstraint**: Tests zone-level topology constraint

**Test Coverage:**
- Constraint behavior at different resource levels (PCS, PCSG, PCLQ)
- Different topology domains (zone, rack, host, none)
- Constraint inheritance and override patterns
- KAI PodGroup structure verification with SubGroups

This PR is part 2 of 4 in the TAS e2e test suite.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

**Dependencies:**
- This PR depends on PR #348 (infrastructure) being merged first

**Test Verification:**
- All tests compile successfully with `-tags e2e`
- Linter passes with 0 issues
- Added 398 lines of test code across 5 test functions
- Added 5 YAML test scenario files

**What's Next:**
- PR #350: Scaling & hierarchical tests (5 tests)
- PR #351: Advanced scenarios & edge cases (4 tests)

**File Summary:**
- Modified: 1 file (topology_test.go - added 5 tests)
- New: 5 YAML test scenario files

#### Does this PR introduce a API change?

```release-note
NONE
```

#### Additional documentation e.g., enhancement proposals, usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: operator/e2e/tests/topology_test.go [hunk 1]
```diff
@@ -213,3 +214,396 @@ func Test_TAS2_MultipleCliquesWithDifferentConstraints(t *testing.T) {
...
+// 1. Deploy workload with full 3-level hierarchy:
+//   - PCS: packDomain=block
+//   - PCSG: packDomain=rack (stricter than block)
+//   - PodCliques (prefill, decode): packDomain=host (strictest)
+//
+// 2. Verify all 8 pods are scheduled successfully
+// 3. Verify all pods are on the same host (strictest constraint wins)
+// 4. Verify constraint inheritance and override behavior
+// Test_TAS3_PCSOnlyConstraint tests constraint only at PCS level with no PCSG/PCLQ constraints
+func Test_TAS3_PCSOnlyConstraint(t *testing.T) {
```
Comment (gflarity) [inline operator/e2e/tests/topology_test.go]: This comment block describes `Test_TAS_SP1_FullHierarchyWithCascadingConstraints, not Test_TAS3_PCSOnlyConstraint. Looks like it's an old comment.

File: operator/e2e/tests/topology_test.go [hunk 2]
```diff
@@ -213,3 +214,396 @@ func Test_TAS2_MultipleCliquesWithDifferentConstraints(t *testing.T) {
...
+	logger.Info("2. Deploy workload (TAS4: PCSG-only constraint)")
+	allPods, err := deployWorkloadAndGetPods(tc, expectedPods)
+	if err != nil {
+		t.Fatalf("Setup failed: %v", err)
+	}
+
+	logger.Info("3. Verify PCSG worker pods (2 total, 1 per replica) in same rack")
+	workerPods := utils.FilterPodsByLabel(allPods, "grove.io/podcliquescalinggroup", "tas-sl-pcsg-only-0-workers")
+	if len(workerPods) != 2 {
+		t.Fatalf("Expected 2 worker pod, got %d", len(workerPods))
```
Comment (gflarity) [inline operator/e2e/tests/topology_test.go]: Minor typo - should be plural "pods".

```suggestion
		t.Fatalf("Expected 2 worker pods, got %d", len(workerPods))
```

File: operator/e2e/tests/topology_test.go [hunk 3]
```diff
@@ -213,3 +214,396 @@ func Test_TAS2_MultipleCliquesWithDifferentConstraints(t *testing.T) {
...
+
+	logger.Info("🎉 TAS5: Host-Level Constraint test completed successfully!")
+}
+
+// Test_TAS6_ZoneLevelConstraint tests PCS with block constraint and standalone PCLQ with host constraint
+// 1. Deploy workload with PCS block constraint and PCLQ host constraint (no PCSG layer)
+// 2. Verify 2 pods on same host (PCLQ constraint, strictest)
+// 3. Verify both pods in same block (PCS constraint inherited)
+// 4. Verify KAI PodGroup has block constraint at top level, 1 SubGroup with host constraint
+func Test_TAS6_ZoneLevelConstraint(t *testing.T) {
```
Comment (gflarity) [inline operator/e2e/tests/topology_test.go]: The comment describes "PCS with block constraint and PCLQ host constraint" but looks like the actual test uses zone-level constraint at PCS level (as seen in `tas-zone-level.yaml` which has `packDomain: zone`).

File: operator/e2e/tests/topology_test.go [hunk 4]
```diff
@@ -213,3 +214,396 @@ func Test_TAS2_MultipleCliquesWithDifferentConstraints(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("Setup failed: %v", err)
+	}
+
+	logger.Info("3. Verify all 4 pods in same rack (inherited from PCS)")
+	if err := utils.VerifyPodsInSameTopologyDomain(tc.Ctx, tc.Clientset, allPods, setup.TopologyLabelRack, logger); err != nil {
+		t.Fatalf("Failed to verify all pods in same rack: %v", err)
+	}
+
+	logger.Info("4. Verify PCSG worker pods (2 total, 1 per replica)")
```
Comment (shayasoolin) [inline operator/e2e/tests/topology_test.go]: I don't understand what steps 4 and 5 provide in terms of topology validation. Seems to me redundant and already tested in more basic grove components.

File: operator/e2e/tests/topology_test.go [hunk 5]
```diff
@@ -213,3 +214,396 @@ func Test_TAS2_MultipleCliquesWithDifferentConstraints(t *testing.T) {
...
+	// Verify top-level TopologyConstraint (PCS level: rack)
+	if err := utils.VerifyKAIPodGroupTopologyConstraint(podGroup, setup.TopologyLabelRack, "", logger); err != nil {
+		t.Fatalf("Failed to verify KAI PodGroup top-level constraint: %v", err)
+	}
+
+	// Verify SubGroups (2 PCLQ children + 1 router standalone = 3 total)
+	// Note: PCSG parent groups are NOT created when PCSG has nil TopologyConstraint (PR #357)
+	expectedSubGroups := []utils.ExpectedSubGroup{
+		// Worker PCLQs (directly under PCS constraint, no PCSG parents)
+		{Name: "tas-sl-pcs-only-0-workers-0-worker", MinMember: 1, Parent: nil},
```
Comment (shayasoolin) [inline operator/e2e/tests/topology_test.go]: Consider having a function for that. I think the name at least should be constructed using a helper function, rather than manually.

File: operator/e2e/tests/topology_test.go [hunk 6]
```diff
@@ -213,3 +214,396 @@ func Test_TAS2_MultipleCliquesWithDifferentConstraints(t *testing.T) {
...
+	logger.Info("3. Verify PCSG worker pods (2 total, 1 per replica) in same rack")
+	workerPods := utils.FilterPodsByLabel(allPods, "grove.io/podcliquescalinggroup", "tas-sl-pcsg-only-0-workers")
+	if len(workerPods) != 2 {
+		t.Fatalf("Expected 2 worker pod, got %d", len(workerPods))
+	}
+	if err := utils.VerifyPodsInSameTopologyDomain(tc.Ctx, tc.Clientset, workerPods, setup.TopologyLabelRack, logger); err != nil {
+		t.Fatalf("Failed to verify worker pods in same rack: %v", err)
+	}
+
+	logger.Info("4. Verify router pods (2 standalone, unconstrained)")
```
Comment (shayasoolin) [inline operator/e2e/tests/topology_test.go]: What does this validation step adds?

File: operator/e2e/tests/topology_test.go [hunk 7]
```diff
@@ -213,3 +214,396 @@ func Test_TAS2_MultipleCliquesWithDifferentConstraints(t *testing.T) {
...
+	// Additional check: verify both pods have same node name
+	if len(allPods) != 2 {
+		t.Fatalf("Expected 2 pods, got %d", len(allPods))
+	}
+	if allPods[0].Spec.NodeName != allPods[1].Spec.NodeName {
+		t.Fatalf("Pods not on same node: %s vs %s", allPods[0].Spec.NodeName, allPods[1].Spec.NodeName)
+	}
+
+	logger.Info("4. Verify KAI PodGroup has correct SubGroups (PCLQ-only host constraint)")
+	podGroups, err := utils.WaitForKAIPodGroups(tc.Ctx, tc.DynamicClient, tc.Namespace, "tas-host-level", tc.Timeout, tc.Interval, logger)
```
Comment (shayasoolin) [inline operator/e2e/tests/topology_test.go]: Take "tas-host-level" from tc (same for all tests in the file, as well as other places in the code e.g. L447, L459)

File: operator/e2e/tests/topology_test.go (hunk 3 above)
Comment (shayasoolin) [inline operator/e2e/tests/topology_test.go]: It's hard to understand what makes this test special compared to Test_TAS3_PCSOnlyConstraint.
It's not the fact that it's zone-level, but the fact that it has topology only on the PCS, like TAS3, but unlike TAS3, its PCLQs are not part of PCSG.
If I got it right, please rename the test and the yaml for clarity. Thanks

File: operator/e2e/utils/naming.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// */
+
+import "fmt"
+
+// GetBasePodGangName constructs the base PodGang name for a specific PodGangSet replica.
```
Comment (shayasoolin) [inline operator/e2e/utils/naming.go]: What is PodGangSet - pgs? I think it's a mistake

Comment (Ronkahn21) [reply operator/e2e/utils/naming.go]: fixed
